test/dataframe.jl

```diff
@@ -1723,6 +1723,30 @@ end
                                     cols=:orderequal)
 end
 
+@testset "vcat with source" begin
+    df1 = DataFrame(A=1:3, B=1:3)
+    df2 = DataFrame(A=4:6, B=4:6)
+    df3 = DataFrame(A=7:9, C=7:9)
+    df4 = DataFrame()
+
+    for col in [:source, "source"]
+        @test vcat(df1, df2, df3, df4, cols=:union, source=col) ≅
```

Comment:
Just so that `CategoricalArray` isn't the only tested type:
```suggestion
        @test vcat(df1, df2, df3, df4, cols=:union, source=col) ≅
            vcat(df1, df2, df3, df4, cols=:union, source=[1, 2, 3, 4]) ≅
```